vyper/ast/nodes.py

```diff
@@ -141,7 +141,34 @@ def _to_dict(value):
     # if value is a Vyper node, convert to a dict
     if isinstance(value, VyperNode):
         return value.to_dict()
-    return value
+
```

Comment:
kind of weird to import type modules here. is there some way we can rethink this structure?

Reply:
i think the real issue here is that the representation of the `_metadata["type"]` field is not implemented. i think it will be cleaner if we add a method to all vyper types which is like "serialize_for_ast" or something (i didn't think very hard about the name, feel free to come up with alternatives).